Added I18n to the app

sebarmano opened the pull request:
- Added :locale field to user to be able to set the language preference on sign up.
- Added filter before action to set up locale for each user preference.
- Added translations to every view (using helpers in the views and locale files to get the translations).

sebarmano committed:
Added language selection to sign up form

sebarmano committed:
Added spanish translation to layouts/application page

sebarmano committed:
Added translation for resetting password and default spanish locales

sebarmano committed:
Added validation for locale in User model

sebarmano committed:
Merge branch 'i18n'

Conflicts:
	app/views/books/_form.html.slim
	app/views/books/show.html.slim
	app/views/books/tags.html.slim
	app/views/layouts/application.html.slim
	db/schema.rb

sebarmano committed:
Merged files from master repository and corrected some syntax errors

sebarmano committed:
Added translations to new views

sebarmano committed:
Added translations to new views

sebarmano committed:
Merge branch 'master' of github.com:sebarmano/freeshelf

Conflicts:
	config/locales/es.yml

sebarmano committed:
Changed some authentication tests to use I18n

sebarmano committed:
Fixed syntax error in new user view

cndreisbach commented on app/views/books/_form.html.slim:
```diff
@@ -1,19 +1,20 @@
 = simple_form_for book, :html => {:multipart => true} do |form|
```
You can more easily do I18n with SimpleForm like so: http://simple-form.plataformatec.com.br/#i18n

cndreisbach commented:
After all this hard work, I hesitate to ask, but can you set the forms up the SimpleForm way with I18n?

sebarmano commented:
Sure! I don't think is too hard. I just need to move the translations from the files were they are now to the simple form yaml files and delete the extra text in forms.

sebarmano committed:
Merge remote-tracking branch 'upstream/master'

Conflicts:
	app/views/books/_form.html.slim
	app/views/layouts/application.html.slim

sebarmano committed:
Added translation books form partial

sebarmano committed:
Added translation for videos, favorites and sorting

sebarmano committed:
Merge remote-tracking branch 'upstream/master'

Conflicts:
	app/controllers/users_controller.rb
	app/views/users/_settings.html.slim
	db/schema.rb

sebarmano committed:
Merge branch 'simpleform-i18n'

Conflicts:
	app/views/favorites/show.html.slim

sebarmano committed:
deleted backup files

sebarmano committed:
Fixed some translation bugs in preferences modal

sebarmano committed:
Merge remote-tracking branch 'upstream/master'

Conflicts:
	app/views/books/_edit_tags_form.html.slim

sebarmano commented:
Done. Now all SimpleForm translations have been changed to SimpleForm way. I also added missing translations for settings and fixed a bug that showed the wrong language in form buttons.
